### PR TITLE
Add form validations to Admin auctions page

### PR DESCRIPTION
**Pull request by jessieay**
* Validate presence of various fields
* Also add DB constraints
* Show errors in one flash if wrong
* Render page to inputs not lost if form errors
* Add selectors for datetime form stuff

* Closes https://github.com/18F/micropurchase/issues/549

Future work:
* Add helper text to form to explain field meanings
* Add client-side validation
* Make date / timepickers fancier

screen shot 2016-05-18 at 10 14 06 am

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -33,7 +33,6 @@ group :test do
   gem 'webmock'
   gem 'json-schema'
   gem 'db-query-matchers'
-  gem 'json-schema'
```
Comment (jessieay) [inline Gemfile]: duplicate

File: app/models/auction.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class Auction < ActiveRecord::Base
+  MAX_START_PRICE = 3500
```
Comment (adelevie) [inline app/models/auction.rb]: shouldn't we instead use `StartPriceThresholds::MICROPURCHASE`?

Comment (jessieay) [inline app/models/auction.rb]: oops merge conflict resolution error - will remove

Comment (jessieay) [inline app/models/auction.rb]: removed at 369eaab

File: app/models/auction.rb [hunk 3]
```diff
@@ -10,8 +12,13 @@ class Auction < ActiveRecord::Base
   # Disable STI
   self.inheritance_column = :__disabled
 
+  validates :delivery_deadline, presence: true
```
Comment (harrisj) [inline app/models/auction.rb]: Are we validating here or only on publish?

File: app/models/auction_parser.rb [hunk 4]
```diff
@@ -1,86 +1,58 @@
...
-
-  def general_attributes
     {
-      type: type,
-      title: title,
-      description: description,
-      summary: summary,
-      github_repo: github_repo,
-      issue_url: issue_url,
+      awardee_paid_status: auction_params[:awardee_paid_status],
```
Comment (harrisj) [inline app/models/auction_parser.rb]: I approve

File: config/locales/simple_form.en.yml [hunk 5]
```diff
@@ -7,3 +7,18 @@ en:
       mark: '*'
     error_notification:
       default_message: "Please review the problems below:"
+    labels:
+      auction:
+        start_datetime: "Start date and time (in UTC)"
```
Comment (harrisj) [inline config/locales/simple_form.en.yml]: :+1:

File: app/models/auction.rb (hunk 3 above)
Comment (jessieay) [inline app/models/auction.rb]: the issue requested it be validated on create: https://github.com/18F/micropurchase/issues/549

do you think we should only validate on publish?

File: app/models/auction_parser.rb (hunk 4 above)
Comment (jessieay) [inline app/models/auction_parser.rb]: ✔️

**Comment by jessieay** [comment]
Updated to only validate some fields on publish. Still want to validate title and a few others on create. :)

File: db/migrate/20160517224513_add_null_false_to_auction_fields.rb [hunk 6]
```diff
@@ -0,0 +1,32 @@
+class AddNullFalseToAuctionFields < ActiveRecord::Migration
+  def up
+    change_column :auctions, :billable_to, :string, default: ''
```
Comment (jessieay) [inline db/migrate/20160517224513_add_null_false_to_auction_fields.rb]: explanation of this: when fields are able to be null, I prefer to set the default to empty string at the DB level. That way, no `nil` errors when we render the obj in the view. :)

**Comment by adelevie** [comment]
Where'd all the `let`s go?!! I miss them.

**Comment by mtorres253** [comment]
closes #592 as well

**Pull request merged**